app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedPresenter.java

```diff
@@ -40,45 +45,78 @@
 
     @Inject
     public NewsFeedPresenter() {
+        disposables = new CompositeDisposable();
     }
 
     @Override
-    public void start() {
+    public void subscribe() {
         Log.d(TAG, "start: " + mainApplication);
         Category.resolveCategory(mainApplication, Arrays.asList(Category.values()));
         initStartValues();
         updateActualNews();
     }
 
+    @Override
+    public void unsubscribe() {
+        disposables.clear();
+    }
+
     @Override
     public void updateActualNews() {
-        view.showProgressBar();
-        updateNewsStorage();
+        NewsFeedParams params = new NewsFeedParams(new Filter(category, showOnlyFavorite));
+        Disposable subscription = updateStorageUseCase.run(params)
+                .flatMap(updatedNews -> getFilteredNewsUseCase.run(params))
+                .doOnSubscribe(disposable -> view.showProgressBar())
+                .onErrorResumeNext(throwable -> getFilteredNewsUseCase.run(params))
+                .subscribeOn(Schedulers.computation())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        resultNews -> { /* Emit */
+                            setCashedNews(resultNews);
+                            updateNewsListUI(resultNews);
+                            view.hideProgressBar();
+                            view.hideRefreshLayout();
+                        },
+                        throwable -> { /* Error */
+                            Log.e(TAG, "updateActualNews failure: " + throwable);
+                            view.hideProgressBar();
+                            view.hideRefreshLayout();
+                            view.showEmptyView();
+                        }
+                );
+        disposables.add(subscription);
     }
 
     @Override
     public void openNewsDetails(int position, Context context) {
-        NewsItem item = news.get(position);
+        NewsItem item = cashedNews.get(position);
         String url = item.getUrl();
         navigator.openWebView(url);
     }
 
     @Override
     public void changeNewsFavoriteState(int position) {
```

Comment:
This method is overengineered. You should simply update an item and pass a fresh list to be rendered. You don't need:
1. `NewsFeedParams` for this use case as this use case needs just a single list to work.
2. All code that resides in `onNext` should be reworked and simply initiate rendering of the list. This would be somewhat similar to what you see in the main Rx chain in case of success.

The rest of the code should be removed.